compiler/libpc300/sc2.c

```diff
@@ -2950,7 +2950,7 @@ static char itohstr[30];
     val>>=4;
   } /* endfor */
   i=max-1;
-  while (nibble[i]==0 && i>0)   /* search for highest non-zero nibble */
+  while (i>0 && nibble[i]==0)   /* search for highest non-zero nibble */
```

Comment:
What is the bug here?

Reply:
i can be -1 ?

Reply:
I don't see how. `i` starts at 7, and when it reaches 0, the condition will fail.